roles/kubernetes-apps/metrics_server/defaults/main.yml

```diff
@@ -0,0 +1,13 @@
+---
+metrics_server_namespace: "kube-system"
```

Comment:
Why do you define `metrics_server_namespace` when you do not use it everywhere? Lots of the yaml files with SA etc has been hardcoded to kube-system. Either use it or remove it